### DISCUSSION TITLE
Decide the Fate of Mod

### DISCUSSION THREAD

**Discussion by vborovin** [sm-augmented/hub#21] [General]
Without Dragonik there is no possibility to develop gameplay changes. And so, next questions arise:

- [x] What should the mod focus on without the design lead? #20 
- [x] #32 
- [x] What technical tasks need to be prioritized? #20

**Comment by vborovin** [comment]
Lupus:
If gameplay changes are not going to be made at all, from a non-technically standpoint I'd love to see any armor or weapons that can be finished added. Other than that, once the mod is returned fully to what it was, and any bug fixes that can be made are made, if you have no intention of either finding a new lead or just making no effort to make any further gameplay improvements or changes, then just leave it once it's there.

Then, if you intend to bring back testers, I would recommend allowing testers to help in proposing changes or such. It wont be as clear of a vision of having a proper lead, but it could at least give you ideas of what could be changed or added. Though I'll still complain about TH DFA until the heat death of the universe

**Comment by vborovin** [comment]
DaF:
imo tbh, the solution is "the last stand update" format
The fate of this mod is simple, the update with all stuff prior showed can be added, but after that, with the actual feedback
Maybe the open format can be the salvation, but it's unclear if a real team would take the position
Gather back the old team and try something together

**Comment by SteelDad40k** [comment]
I think it's worth asking the question, what exactly do we want to see in SMA. New armor, weapons or maybe a new unique mode with unique classes?

**Comment by vborovin** [comment]
> I think it's worth asking the question, what exactly do we want to see in SMA. New armor, weapons or maybe a new unique mode with unique classes?

Yeah, what is the goal of mod now? Do we still keep all those plans we had before, even without the person who was responsible for designing them?

**Comment by SteelDad40k** [comment]
Unfortunately, I don’t know what the plans were regarding the mod, but I think it’s worth implementing what was known, and then already thinking about what will happen next.

**Comment by vborovin** [comment]
> Unfortunately, I don’t know what the plans were regarding the mod, but I think it’s worth implementing what was known, and then already thinking about what will happen next.

There were plans for a very long shot and included:

- Focus on PvE mostly with PvP adapted to PvE basis given free time
- 7 new classes
- Numerous new weapons and perks, finishing changes for existing weapons and perks to align them into one system, including but not limited to:
  - 2H/DW versions of melee weapons
  - 3 shields
  - Phobos and Tigrus versions of BP and StoB
  - New equipment
  - God-dedicated perks for chaos and their mirrors for loyals
- AI improvements
- Numerous QoL changes, including UI and animation changes and fixes
- New maps' setups, including but not limited to:
  - Infinite horde mode
  - Revamped arenas, each lasts full 20 waves and dedicated to specific enemy side
- And more?

We may keep this at the core of SMA and this will be enough to develop for a very long timespan. But this will require insane amount of design thinking, as all details are gone with Dragonik.

**Reply by SteelDad40k** [reply]
then I propose to pay attention to those items that do not require design solutions Dragonik. Are there any at all?

**Reply by vborovin** [reply]
Yeah, there are, more or less:
- Numerous QoL changes, including UI and animation changes and fixes
- New maps' setups
- New armor pieces and accessories

At least those. Unknown what to do with classes, weapons, perks, enemies, etc.

**Reply by vborovin** [reply]
Plan moved to separate page: #20

**Comment by vborovin** [comment]
> How the ground, methodology, organization and architecture need to evolve?

Design docs:
- Big design chunks (classes, weapon+perks, etc) are described via design docs, that contain all necessary stats and solutions required to implement stuff ingame.
- Proposed changes to the game described via issues. Issue contains argumentation and possible solutions, their drawbacks, unresolved questions (mini-RFC?). Some initial values are taken to start the development, based on existing design docs and change's future place within them (role of some new thing, changes to existing docs, necessity to write new design doc to describe some new big thing). 
- When issue and it's initial values are accepted after a reviewal and corrections, it can be implemented. Then, they are checked by testers and feedback is used to iteratively find optimal local extremum. Issue changes along each stated optimization step.  After acceptance from testers and the team (?) finishing version of issue is what will be committed to the mod and described in design docs. After that issue is closed.

Alternatives:
- Another option is that proposed change is directly made in RFC-like manner via PRs. When proposed design docs changes are accepted by the community, an issue calling to implement the proposed change is created, planned and implemented. Then corrected as previously if needed on each testing/optimization step. After the change is accepted ingame, PR and issue are closed.

Mod goals and values:
- Mod need to balance between "pro" and "casual" ways, allowing both.
- Mod need to tend to be coherent, aiming to minimize content and community splitting (e.g. no separate PvP, PvE, demo mode, etc)
- Planning, update cycle/iterations ???
- TBC

**Reply by vborovin** [reply]
Detailed tasks were created for the question: #32 
Mod values moved to: #20 
Planning tasks moved to: #35

**Comment by vborovin** [comment]
> What technical tasks need to be prioritized?
> What should the mod focus on without the design lead?

Proposed solution:
Until the initial design vision is gained and formalized via design docs, only technical tasks can be worked on. Which include:
- Fix any starting issues and restore pre-deletion state
- Define how the ground, methodology, organization and architecture need to evolve. 
- Define planning process, include next things into it, and start using the [plan](https://github.com/sm-augmented/hub/discussions/20)
- Numerous QoL changes, including UI and animation changes and fixes: customizer changes, public search fix, online showing, etc.
- New maps' setups
- New armor pieces, accessories and other visual mostly related content (colors, badges, icons, locales, etc)
- Slot issue exploration
- Launcher stabilization
- Game launch stabilization (new and old launch mode unification)
- Chunk and package migration, finding a way to expand ingame modes in a single game instance

New design vision must be gained and formalized in parallel with any other technical tasks.

**Reply by vborovin** [reply]
Plan moved to separate page: #20 
Planning tasks moved to: #35
